components/common/reconcilehelper/util.go

```diff
@@ -211,9 +213,9 @@ func CopyVirtualService(from, to *unstructured.Unstructured) bool {
 		return true
 	}
 
-	requiresUpdate := !reflect.DeepEqual(fromSpec, toSpec)
-	if requiresUpdate {
+	requireUpdate := !reflect.DeepEqual(fromSpec, toSpec)
```

Comment:
should we use !equality.Semantic.DeepDerivative instead?

Reply:
for VirtualService, there is  no default value which can afftect DeepEqual, so in my opinion there no need here